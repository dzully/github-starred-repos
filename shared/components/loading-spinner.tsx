import React from 'react'
import { motion } from 'framer-motion'
import { cn } from '@/lib/utils'

interface LoadingSpinnerProps {
  size?: 'sm' | 'md' | 'lg'
  className?: string
}

const sizeMap = {
  sm: 'w-4 h-4 border-2',
  md: 'w-8 h-8 border-3',
  lg: 'w-12 h-12 border-4',
}

export const LoadingSpinner = ({
  size = 'sm',
  className,
}: LoadingSpinnerProps) => {
  return (
    <div
      data-testid="loading-spinner"
      className={cn('flex justify-center items-center', className)}
    >
      <motion.div
        className={cn(
          'rounded-full border-primary/30 border-t-primary',
          sizeMap[size]
        )}
        initial={{ rotate: 0 }}
        animate={{ rotate: 360 }}
        transition={{
          duration: 0.8,
          repeat: Infinity,
          ease: 'linear',
        }}
      />
    </div>
  )
}
