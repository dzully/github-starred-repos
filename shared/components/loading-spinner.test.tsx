import React from 'react'
import { describe, expect, it } from 'vitest'
import { render, screen } from '@testing-library/react'
import { LoadingSpinner } from './loading-spinner'
import renderer from 'react-test-renderer'

describe('LoadingSpinner', () => {
  it('renders the loading spinner', () => {
    render(<LoadingSpinner />)
    const spinner = screen.getByTestId('loading-spinner')
    expect(spinner).toBeInTheDocument()
  })

  it('applies the correct classes for styling', () => {
    render(<LoadingSpinner />)
    const spinner = screen.getByTestId('loading-spinner')
    expect(spinner).toHaveClass('flex justify-center items-center h-full')
  })

  it('applies animation properties', () => {
    render(<LoadingSpinner />)
    const spinner = screen.getByTestId('loading-spinner')
    expect(spinner).toHaveClass('flex justify-center items-center h-full')
  })

  it('matches snapshot', () => {
    const tree = renderer.create(<LoadingSpinner />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
