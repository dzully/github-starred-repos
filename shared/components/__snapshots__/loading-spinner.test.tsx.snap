// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner > matches snapshot 1`] = `
<div
  className="flex justify-center items-center h-full"
  data-testid="loading-spinner"
>
  <div
    className="w-16 h-16 border-4 border-primary border-t-transparent rounded-full"
    style={{}}
  />
</div>
`;
